boa_parser/src/parser/expression/left_hand_side/template.rs

```diff
@@ -77,13 +79,19 @@ where
                     )?;
                 }
                 TokenKind::TemplateNoSubstitution(template_string) => {
+                    let mut hasher = FxHasher::default();
+                    hasher.write_u32(start_position.line_number());
+                    hasher.write_u32(start_position.column_number());
+                    hasher.write_usize(cursor.identifier());
```

Comment:
There is a very small but possible possibility that it would have that same hash as a previous `TaggedTemplate`.

A solution would be to divide the `u64` identifier into two parts (`u32: count`, `u32: generation`), where the `count` starts from `0` and increments, and the generation is passed from the context as we do now.

What do you think?

Reply:
Sounds good. I switched to a count and also found some more places where we have to increment the parser id.